domain/secret/service/exportimport.go

```diff
@@ -10,13 +10,14 @@ import (
 
 	coremodel "github.com/juju/juju/core/model"
 	coresecrets "github.com/juju/juju/core/secrets"
+	"github.com/juju/juju/domain"
 	"github.com/juju/juju/domain/secret"
 )
 
 // GetSecretsForExport returns a result containing all the information needed to
 // export secrets to a model description.
 func (s *SecretService) GetSecretsForExport(ctx context.Context) (*SecretExport, error) {
-	secrets, secretRevisions, err := s.secretState.ListSecrets(ctx, nil, nil, secret.NilLabels)
+	secrets, secretRevisions, err := s.secretState.ListAllSecrets(ctx)
```

Comment:
This entire operation still isn't atomic though. We use separate txns for
- secrets
- secret grants
- secret consumers
- secret remote consumers